Announce virtual comp schedule (#58)
srobo/team-emails: SR2020/2020-07-05-virtual-competition-timings.md

---

> to: sr2020-teams

Author: Only Virtual comp teams?

Reviewer: ```suggestion
to: Virtual competition signups
```

subject: SR2020 Virtual Competition Schedule
---

Hi all

We're pleased to announce the structure of the SR2020 virtual competition. We will be running 5 sessions over 3 weekends.

> You will have until 10am on each day to make your changes before your team's code is frozen. Then **at 3pm we will be live-streaming matches** back-to-back on YouTube.

Reviewer: I think we should document here *how* submission will work. Namely, saying we'll take a project on the IDE named `virtual-competition`, and download it at 10am.

Reviewer 2: There's a couple of considerations around this which we'd not previously discussed that lead to us being less confident about an IDE solution. I think it could still be a valid fallback, but IMO leaving this vague for now gets us more time to find an alternative (I'm hoping to get my submitter thing working this weekend).

I'll put the discussion details on https://github.com/srobo/tasks/issues/593.

Author: I think it's worth mentioning that the plan is to let them know later

**At 3pm we will be live-streaming matches** back-to-back on YouTube.

| Date           | Competition Stage                                                   |
|----------------|---------------------------------------------------------------------|
| 11th July 2020 | [League Part 1](https://studentrobotics.org/events/sr2020/virtual-competition-league-1/)       |
| 12th July 2020 | [League Part 2](https://studentrobotics.org/events/sr2020/virtual-competition-league-2/)       |
| 18th July 2020 | [League Part 3](https://studentrobotics.org/events/sr2020/virtual-competition-league-3/)       |
| 19th July 2020 | [League Part 4](https://studentrobotics.org/events/sr2020/virtual-competition-league-4/)       |
| 25th July 2020 | [Knockouts and Final](https://studentrobotics.org/events/sr2020/virtual-competition-knockouts/)|

## Version 0.3.1 of the Simulation

We have released version 0.3.1 of the competition simulation which you can download from the [Docs page](https://studentrobotics.org/docs/competition-simulator/).

Here are some of the notable changes since 0.2.0:

> - Raise an error if you try to `R.sleep()` with negative seconds

Reviewer 2: There was actually another change which also errors for `0`, so we probably should include that,
```suggestion
- Raise an error if you try to `R.sleep()` with seconds less than or equal to 0
```

Author: Why error for 0? Seems like it should just continue

Reviewer 2: It was part of the PR which fixed an issue where sleeping for small times was rounded to zero and that made it too easy to end up with a stopped simulator. Webots doesn't advance time if you step for zero duration.

Since I actually changed the approach in that PR to round up the sleep time we maybe could now allow zero again, however I hope that won't actually impact anyone.

In any case, disallowing zero is what 0.3.1 does and is worth documenting.

> - Fix rounding errors in vision code

Reviewer 2: Which change is this referring to? Is this the float rounding errors issue?
I'm concerned that teams may think this refers to the token distance issue reported in the forum, which I don't think has been fixes.

Assuming this is the floating point rounding issue, perhaps say:
```suggestion
- Fix intermittent `ValueError`s in vision code caused by floating point rounding
```

> - Only launch one Webots thread when using custom initialisation

Reviewer 2: I'd be tempted to omit this from the email. I'm not aware that any team has been impacted by this and I doubt this bullet will mean much to them.

- Stop Python processes being left behind after running the simulation
- Fix Webots access violations (Controller exited with status -1)
- Create a log file for each robot to aid debugging

> 

Reviewer 2: I think it would be good to also include a mention of the new `Robot` variant we introduced that allows for more deterministic robot behaviour. I don't know if we've written the docs for it yet, it would be great to be able to link to them.

Perhaps add something like this:
``` markdown
* Added [opt-in support](<docs-link>) for more control over
  the progression of simulation time, which may offer more
  deterministic robot behaviour
```

Author: 👍Will wait for the docs change before adding this

Reviewer 2: Docs PR: https://github.com/srobo/docs/pull/139

Reviewer: @antoinepetty this has now merged, would be good to include

Author: > @antoinepetty this has now merged, would be good to include

Already added :)

|----------------|---------------------------------------------------------------------|
| 11th July 2020 | [League Part 1](https://studentrobotics.org/events/sr2020/virtual-competition-league-1/)       |
| 12th July 2020 | [League Part 2](https://studentrobotics.org/events/sr2020/virtual-competition-league-2/)       |
| 18th July 2020 | [League Part 3](https://studentrobotics.org/events/sr2020/virtual-competition-league-3/)       |
| 19th July 2020 | [League Part 4](https://studentrobotics.org/events/sr2020/virtual-competition-league-4/)       |
| 25th July 2020 | [Knockouts and Final](https://studentrobotics.org/events/sr2020/virtual-competition-knockouts/)|

## Version 0.3.1 of the Simulation

We have released version 0.3.1 of the competition simulation which you can download from the [Docs page](https://studentrobotics.org/docs/competition-simulator/).

Here are some of the notable changes since 0.2.0:

- Raise an error if you try to `R.sleep()` with negative seconds
- Fix rounding errors in vision code
- Only launch one Webots thread when using custom initialisation
- Stop Python processes being left behind after running the simulation

> - Export 3D HTML recordings of matches in competition mode

Reviewer: This probably isn't necessary to inform teams about, just because they won't be running things in comp mode

- Fix Webots access violations (Controller exited with status -1)
- Create a log file for each robot to aid debugging

> - End support for Python 3.5

Reviewer: This isn't quite true. We still support it, it's just the release process no longer uses it
